pandas/core/indexes/interval.py

```diff
@@ -766,8 +766,11 @@ def get_loc(self, key, method=None):
                 key = Interval(left, right, key.closed)
             else:
                 key = self._maybe_cast_slice_bound(key, 'left', None)
-
-            start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            try:
+                start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            except TypeError:
+                # get_loc should raise KeyError
+                raise KeyError('key is hashable but of incorrect type')
```

Comment:
I think for key errors we typically just pass the key itself as message, so might be good to be consistent with that. Or at least I would include the key in the message, something like: `"Key {0} is of the incorrect type".format(key)`